src/options-processing.cc

```diff
@@ -27,32 +27,35 @@ const char * service_dir_opt::get_user_home()
 
 void service_dir_opt::build_paths(bool am_system_init)
 {
-    /* service directory name */
-    if (service_dir == nullptr && ! am_system_init) {
-        const char * user_home = get_user_home();
-        if (user_home != nullptr) {
-            size_t user_home_len = strlen(user_home);
-            size_t dinit_d_len = strlen("/dinit.d");
-            size_t full_len = user_home_len + dinit_d_len + 1;
-            char *service_dir_w = new char[full_len];
-            std::memcpy(service_dir_w, user_home, user_home_len);
-            std::memcpy(service_dir_w + user_home_len, "/dinit.d", dinit_d_len);
-            service_dir_w[full_len - 1] = 0;
-
-            service_dir = service_dir_w;
-            service_dir_dynamic = true;
+    if (service_dirs.empty()) {
+        bool home_service_dir_set = false;
+
+        /* service directory name */
+        if (not am_system_init) {
```

Comment:
For consistency prefer `!` instead of `not`
```suggestion
        if (!am_system_init) {
```